### DISCUSSION TITLE
option to output all intermediate steps up to the sampling steps amount entered as images

### DISCUSSION THREAD

**Discussion by OWKenobi** [hlky/stable-diffusion#238] [Ideas]
Hello,

for some images, the result changes drastically from one sampling step to the other, and one time, 14 is perfect but 15 is a desaster, sometimes the sweet spot is on 20, and so on. Now, instead of having to manually produce all sampling outputs for comparison, which takes a long time, and the computer re-calculates all the early steps all the time, could you add an option so that _all intermediate steps up to the sampling steps amount entered, will be outputted as images?_ 

This way, if you select 20 sampling steps, in the time of only 1 generation you would create 20 variations of your image.

**Comment by Baku-Rue** [comment]
Was just getting on here to request this same exact thing. Being able to get even a range of pictures from a set of steps would be amazing. I was considering trying to loop a prompt or even use the multi prompt feature, however you can only set the steps by the slider in the GUI and not in the actual prompt itself.

**Reply by Baku-Rue** [reply]
I actually just found this repo from another post asking for something similar. Its a little janky but it does exactly what I was looking for.
https://github.com/404james/stable-diffusion-grids

**Comment by OWKenobi** [comment]
Hello Baku-Rue,

awesome, this looks like an automated batch maker. Thanks for the tip, I'll
have a look!

Sadly, it doesn't catch the idea that, in theory, the diffusion would have
to run only ONCE to generate ALL steps, which is extremely faster.


Am Fr., 2. Sept. 2022 um 22:11 Uhr schrieb Baku-Rue <
***@***.***>:

**Comment by Lumoria** [comment]
This is much needed! Ideally, it should be implemented like how it worked in disco diffusion. Have the diffusion process output an intermediate frame for every X number of steps, X being a variable set by the user.

**Comment by cobryan05** [comment]
This would actually be a fairly simple addition to https://github.com/hlky/stable-diffusion-webui/pull/611 . I might look at it tonight.

**Reply by cobryan05** [reply]
Added to PR
